src/compiler/checker.ts

```diff
@@ -18106,17 +18110,18 @@ namespace ts {
 
                     forEach(overloads, o => {
                         const deviation = getEffectiveDeclarationFlags(o, flagsToCheck) ^ canonicalFlags;
+                        const name = getNameOfDeclaration(o);

```

Comment:
Despite the repetition, calling `getNameOfDeclaration(o)` everywhere we had `o.name` is more efficient since we don't look up the name if there are no deviations.